contracts/Pool.sol

```diff
@@ -102,18 +110,20 @@ contract Pool {
     /// @notice Returns amount of received funds available for collection
     /// by the sender of the message
     /// @return collected The available amount
-    function collectable() public view returns (uint256) {
+    function collectable() public view returns (uint128) {
         Receiver storage receiver = receivers[msg.sender];
         uint64 collectedCycle = receiver.nextCollectedCycle;
         if (collectedCycle == 0) return 0;
+        uint256 currFinishedCycle = block.number / cycleBlocks;
+        if (collectedCycle > currFinishedCycle) return 0;
         uint256 collected = 0;
         int256 lastFundsPerCycle = receiver.lastFundsPerCycle;
```

Comment:
Probably makes sense to have this as uint128, too.

Reply:
Fixed